integration/ca/ca.go

```diff
@@ -79,7 +79,9 @@ func (ca *CA) WriteCertificate(template *x509.Certificate, certPath string, keyP
 		return err
 	}
 
-	if err := writeExclusivePEMFile(keyPath, "PRIVATE KEY", 0600, keyBytes); err != nil {
+	// TODO: private keys should not be worldreadable. This is required when
```

Comment:
Should be able to solve this by making them just readable by the Cortex user, right? `root` will still have access anyway.

Reply:
That is a file created by `go test` by your development user account so using that uid, so you won't be able to chown it without sudo.